tests/tests/test_demo_artifact.py

```diff
@@ -69,7 +71,15 @@ def run_demo_script_up():
                     self.auth.password = password
                     assert len(password) == 12
                     break
-            return proc
+            # Make sure that there are no errors in the script
+            if demo_script.poll() != None:
+                logging.error(
+                    "The demo script could not be brought up: error code: %d" %
+                    demo_script.returncode)
+                raise DemoScriptException(demo_script.returncode)
+            assert len(password) == 12, \
+                "The demo-script failed to initialize the environment for the tests"
```

Comment:
Good idea to check this here. I suggest to remove the other assert in line 72 to have the test more tidy